0005-configuration-management-extension/README.md

```diff
@@ -3,102 +3,151 @@ name: Configuration Management Extension
+  It is RECOMMENDED to omit the `<config_format>` resource path suffix in case the desired configuration format matches the message format.
 
-### Formats
-#### Schemeless JSON
-##### Configuration pull
-Endpoint must send messages to the next topic in order to obtain configuration:
-```
-<endpoint_token>/pull/json
-```
+Example configuration pull request extension-specific resource path parts:
+  - `<endpoint_token>/pull/json`
+  - `<endpoint_token>/pull/json/json`
+  - `<endpoint_token>/pull/json/avro`
 
-The payload should be a JSON-encoded object with the following fields:
-- `id` (required) - id of the message. Should be either string or number. Used to match server response to the request.
-- `configVersion` (optional) - current endpoint configuration version. Should be of integer type.
-- `requiredConfigVersion` (optional) - endpoint configuration version that is requested by endpoint. This provides ability to re-send configuration for endpoints that cannot persist the configuration and provides mechanism to request previous configuration versions (roll-back).
-If there's no `requiredConfigVersion` field in message, then server should use `configVersion` to determine is it needed to send response with configuration, server must not send configuration record if latest available configuration version equals to the one sent by endpoint. If `configVersion` field was excluded from the message, then server will send configuration anyway
+This document only defines the `json` message and configuration formats.
+More formats may be specified in separate RFCs.
+
+When the `<message_format>` is set to `json`, the request payload MUST be a JSON-encoded object with the following [JSON schema](http://json-schema.org/) ([separate file](./config-pull-request.schema.json)):
 
-Example:
 ```json
 {
-  "id": 42,
-  "configVersion": 1,
-  "requiredConfigVersion": 2
+  "$schema": "http://json-schema.org/schema#",
+  "title": "5/CMX configuration pull request schema",
+
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": [ "string", "number" ],
+      "description": "ID of the message used to match server response to the request."
+    },
+    "configId": {
+      "type": "string",
+      "description": "Identifier of the currently applied configuration. Optional."
+    }
+  },
+  "required": [ "id" ],
+  "additionalProperties": false
 }
+
 ```
-Example 2:
+
+Example 1:
 ```json
 {
   "id": 42,
-  "configVersion": 3,
-  "requiredConfigVersion": 2
+  "configId": "97016dbe8bb4adff8f754ecbf24612f2"
 }
 ```
-Example 3:
+
+Example 2:
 ```json
 {
-  "id": 42,
-  "configVersion": 1,
+  "id": 42
 }
 ```
-Example 4:
+
+#### Configuration pull response
+
+The server MUST respond to the configuration pull request by publishing the response message to the `/status` resource path according to the [request/response design pattern defined in 1/KP](/0001-kaa-protocol/README.md#requestresponse-pattern).
```

Comment:
Agree. I'll have to improve the /0001-kaa-protocol/README.md#requestresponse-pattern section to make this possible.